### DISCUSSION TITLE
Issue with Inconsistent Vector Search Results and Geo-Filtering

### DISCUSSION THREAD

**Discussion by diegoanvilla** [qdrant/qdrant#5636] [General]
I am experiencing an issue with Qdrant's vector search functionality when using geo-filtering. Specifically, I am encountering inconsistent search results when filtering by geolocation and applying a score threshold.

I have a collection of documents in Qdrant with vector embeddings and geolocation data. When performing a vector search with a geo-radius filter and a score threshold, the results I get are few and have low scores or at least lower than expected.

Because I knew that high score documents were missing I included an additional filter by document ID of a document i knew was within the radius of the coords provided and has a high score, the desired document appeared with a high score (0.999). However, when I remove the ID filter, the search returns less relevant documents and excludes the desired document, which should have a higher score based on the vector similarity and geolocation proximity.

Here are my queries (leaving out the query field with the vector):

**With document id:**
`{
  "limit": 3000,
  "using": "vector_name",
  "score_threshold": 0.8,
  "with_payload": true,
  "filter": {
    "must": [
      {
        "key": "location",
        "geo_radius": {
          "center": {
            "lon": -xxxxxx,
            "lat": xxxxx
          },
          "radius": 30000
        }
      },
      {
        "key": "id",
        "match": { "any": [100291] }
      }
    ]
  }
}`

**Without document id:**
`{
  "limit": 3000,
  "using": "vector_name",
  "score_threshold": 0.8,
  "with_payload": true,
  "filter": {
    "must": [
      {
        "key": "location",
        "geo_radius": {
          "center": {
            "lon": -xxxx,
            "lat": xxxx
          },
          "radius": 30000
        }
      }
    ]
  }
}
`

**Comment by diegoanvilla** [comment]
After further testing i found that if I edit the payload of the documents that should appear it then shows as a result in a consisten way, this feels like a bug, or is there something I'm missing? Can it be related to the fact that I indexed the location field?

**Reply by diegoanvilla** [reply]
Another thing i just tried, i removed the index of the location field and things seem to be back to normal, odd behaviour but maybe I indexed the field when I shouldn't have or missed something in the docs

**Reply by timvisee** [reply]
Indexing the field that holds the geo data is actually what we would normally recommend. If you have a proper payload index configured, it'll be incorporated in the HNSW graph which should improve search quality.

Please note that these searches are approximate by design and that they may not give 100% accurate results all the time. Though, in your case it sounds like results are off by a big margin.

I'd recommend to play with the HNSW parameters a bit, as that'll allow you to improve search quality.

Could you share your collection configuration? Maybe there's something obvious in there. You can get it by using `GET /collections/my_collection`

**Reply by diegoanvilla** [reply]
Hi! thanks for the reply, after removing the index everything worked fine, the database admin site is recommending me to add the index back again, but I fear that the results from the searcher will be afected negatively again, here is my collection configuration:

{
  "result": {
    "status": "green",
    "optimizer_status": "ok",
    "indexed_vectors_count": 401523,
    "points_count": 194071,
    "segments_count": 8,
    "config": {
      "params": {
        "vectors": {
          "xx": {
            "size": 1536,
            "distance": "Cosine"
          },
          "xx": {
            "size": 1536,
            "distance": "Cosine"
          },
          "xx": {
            "size": 1536,
            "distance": "Cosine"
          }
        },
        "shard_number": 1,
        "replication_factor": 1,
        "write_consistency_factor": 1,
        "on_disk_payload": true
      },
      "hnsw_config": {
        "m": 16,
        "ef_construct": 100,
        "full_scan_threshold": 10000,
        "max_indexing_threads": 0,
        "on_disk": false
      },
      "optimizer_config": {
        "deleted_threshold": 0.2,
        "vacuum_min_vector_number": 1000,
        "default_segment_number": 0,
        "max_segment_size": null,
        "memmap_threshold": null,
        "indexing_threshold": 20000,
        "flush_interval_sec": 5,
        "max_optimization_threads": null
      },
      "wal_config": {
        "wal_capacity_mb": 32,
        "wal_segments_ahead": 0
      },
      "quantization_config": null,
      "strict_mode_config": {
        "enabled": false
      }
    },
    "payload_schema": {
      "id": {
        "data_type": "integer",
        "points": 194071
      },
      "user_id": {
        "data_type": "text",
        "points": 194071
      }
    }
  },
  "status": "ok",
  "time": 0.000140807
}
